moveit_simple/src/moveit_simple.cpp

```diff
@@ -219,16 +217,115 @@ bool Robot::getJointSolution(const Eigen::Affine3d &pose, double timeout,
+    local_seed =  getJointState();
+  }
+
+  return getIK(custom_frame_goal_pose, local_seed, joint_point, timeout);
+}
+
+
+Eigen::Affine3d Robot::transformPoseBetweenFrames(const Eigen::Affine3d &target_pose,
+                                              const std::string& frame_in,
+                                              const std::string& frame_out) const
+{
+
+  if(frame_in.compare(frame_out) == 0 && 
+    robot_model_ptr_->hasLinkModel(frame_in) &&
+    robot_model_ptr_->hasLinkModel(frame_out)) {
+
+      ROS_INFO_STREAM("Returning same target_pose");
+
+      return target_pose;
+  }
+
+  //if( tf_buffer_.canTransform(joint_group_->getEndEffectorName(), ros::Time::now(), ros::Duration(0.1)) ) {
+    try
+    {
+        ROS_INFO_STREAM("Looked up tf named frame: " << frame_in);
+
+        geometry_msgs::PoseStamped pose_stamped_buffer;
+        geometry_msgs::PoseStamped transformed_pose;
+        geometry_msgs::TransformStamped trans_msg;
+        Eigen::Affine3d target_pose_buffer = target_pose;
+
+        trans_msg = tf_buffer_.lookupTransform(frame_out,
+                                               frame_in, 
+                                               ros::Time::now(), 
+                                               ros::Duration(5.0));
+
+        pose_stamped_buffer.header = trans_msg.header;
+        tf::poseEigenToMsg(target_pose, pose_stamped_buffer.pose);
+        
+        tf2::doTransform(pose_stamped_buffer, transformed_pose, trans_msg);
+
+        tf::poseMsgToEigen(transformed_pose.pose, target_pose_buffer);
+
+        ROS_INFO_STREAM("Using TF to lookup transform " << frame_out << " frame: " << std::endl << target_pose_buffer.matrix());
+
+        return target_pose_buffer;
+    }
+    catch (tf2::TransformException &ex)
+    {
+        ROS_ERROR_STREAM("TF transform lookup failed: " << ex.what());
+        throw ex;
+    }
+  //}  
+}
+
 
 bool Robot::getPose(const std::vector<double> & joint_point,
-                                  Eigen::Affine3d & pose) const
+                    Eigen::Affine3d & pose) const
 {
   std::lock_guard<std::recursive_mutex> guard(m_);
 
   return getFK(joint_point, pose);
 }
 
 
+bool Robot::getPose(const std::vector<double> & joint_point,
+                    const std::string& custom_tool_frame,
+                    Eigen::Affine3d & pose) const
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  
+  Eigen::Affine3d pose_buffer;
+
+  // Ideally "tool0" should be replaced with joint_group_->getEndEffectorName()
+  std::string moveit_tool_link = "tool0";
+
+  if(getFK(joint_point, pose_buffer) == true)  
+  {
+    // Transform Target/Goal Point from moveit_end_link to custom tool frame
+    pose = transformPoseBetweenFrames(pose_buffer, moveit_tool_link, 
+                                      custom_tool_frame);
+
+    return true;
+  }                        
+  else  
+  {
+    return false;  
```

Comment:
If you return a failure, you should log an error.  The only exception is if the method is expected to return false (i.e. like a reach check).